mypy/checkexpr.py

```diff
@@ -2960,6 +2960,8 @@ def visit_index_with_type(self, left_type: Type, e: IndexExpr,
         elif (isinstance(left_type, CallableType)
               and left_type.is_type_obj() and left_type.type_object().is_enum):
             return self.visit_enum_index_expr(left_type.type_object(), e.index, e)
+        elif isinstance(left_type, TypeVarType):
```

Comment:
I am sure it cannot cause infinite recursion, because `.upper_bound` cannot be another `TypeVar`.